fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryIntersectionPlan.java

```diff
@@ -202,8 +203,16 @@ public boolean equalsWithoutChildren(@Nonnull RelationalExpression otherExpressi
                comparisonKey.equals(other.comparisonKey);
     }
 
-    @SuppressWarnings("EqualsWhichDoesntCheckParameterClass")
+    @Nonnull
+    @Override
+    public AvailableFields getAvailableFields() {
+        return AvailableFields.intersection(quantifiers.stream()
```

Comment:
`RecordQueryIntersectionPlan` requires at least two children (see line 294). However, I've added a check to `AvailableFields` so that it can't be misused.